packages/test/local-server-stress-tests/src/stressDataObject.ts

```diff
@@ -0,0 +1,320 @@
+								...entry,
+								handle,
+							});
+							break;
+						case "stressDataObject":
+							assert(maybe?.StressDataObject !== undefined, "must be stressDataObject");
+
+							globalObjects.push({
+								type: "stressDataObject",
+								tag: entry.tag,
+								handle,
+								stressDataObject: maybe.StressDataObject,
+							});
+							break;
+						default:
+					}
+				}
+			}
+		}
+		return globalObjects;
+	}
+
+	protected override async getDefaultStressDataObject(): Promise<DefaultStressDataObject> {
+		return this;
+	}
+
+	/**
+	 * this map is special, and doesn't participate in stress. it holds data
+	 * about the name of container objects which have been created. these created objects
+	 * may or may not be attached and be available
+	 */
+	private containerObjectMap: ISharedMap = makeUnreachableCodePathProxy("containerObjectMap");
+	protected async initializingFirstTime(props?: any): Promise<void> {
+		await super.initializingFirstTime(props);
+		this.containerObjectMap = SharedMap.create(this.runtime, "containerObjectMap");
+		this.containerObjectMap.bindToContext();
+
+		this.registerLocallyCreatedObject({
+			type: "stressDataObject",
+			handle: this.handle,
+			tag: `datastore-0`,
+			stressDataObject: this,
+		});
+	}
+
+	protected async initializingFromExisting(): Promise<void> {
+		this.containerObjectMap = (await this.runtime.getChannel(
+			"containerObjectMap",
+		)) as any as ISharedMap;
+	}
+
+	public registerLocallyCreatedObject(obj: ContainerObjects) {
+		if (obj.handle) {
+			const handle = toFluidHandleInternal(obj.handle);
+			if (this.containerObjectMap.get(handle.absolutePath) === undefined) {
+				this.containerObjectMap.set(handle.absolutePath, { tag: obj.tag, type: obj.type });
+			}
+		}
+		this._locallyCreatedObjects.push(obj);
+	}
+}
+
+export const createRuntimeFactory = (): IRuntimeFactory => {
+	const defaultStressDataObjectFactory = new DataObjectFactory(
+		"DefaultStressDataObject",
+		DefaultStressDataObject,
+		[...ddsModelMap.values()].map((v) => v.factory),
+		{},
+		[[StressDataObject.factory.value.type, StressDataObject.factory.value]],
+	);
+
+	const runtimeOptions: IContainerRuntimeOptionsInternal = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				maxOps: 3,
+				initialSummarizerDelayMs: 0,
+			} as any,
+		},
+	};
+
+	return {
+		get IRuntimeFactory() {
+			return this;
+		},
+		instantiateRuntime: async (context, existing) => {
+			return loadContainerRuntime({
+				context,
+				existing,
+				runtimeOptions,
+				registryEntries: [
+					[
+						defaultStressDataObjectFactory.type,
+						Promise.resolve(defaultStressDataObjectFactory),
+					],
+					[
+						StressDataObject.factory.value.type,
+						Promise.resolve(StressDataObject.factory.value),
+					],
+				],
+				provideEntryPoint: async (rt) => {
+					const maybeDefault = await rt.getAliasedDataStoreEntryPoint(
+						DefaultStressDataObject.alias,
+					);
+					if (maybeDefault === undefined) {
```

Comment:
Any particular reason to do an alias race here rather than just create in the detached state?

Reply:
this will all happen when detached. i'm not sure of another way to do it,

Reply:
I think it only happens when detached if the entry point is accessed while detached.  If attach happens prior to accessing the entryPoint then it's a race.

You can do the creation in `instantiateRuntime` (outside of `provideEntryPoint`) when `!existing`, this is how most of our examples do it, aqueduct, etc.

Reply:
Most of those examples likely pre-date entrypoint. i think i actually prefer this pattern, as it doesn't require using variables before they are initialized, which the other pattern does. I think it would be reasonable to make getting the entrypoint part of the attach flow to ensure at least basic initialization happens at least once. we do something similar with datastores that ensures they are realized before attach.